sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreReader.java

```diff
@@ -269,9 +269,11 @@ public StoreReader(
                 hasGoneException.v = hasGoneException.v || (srr.isGoneException && !srr.isInvalidPartitionException);
 
                 if (resultCollector.size() >= replicaCountToRead) {
-                    if (hasGoneException.v && !entity.requestContext.performedBackgroundAddressRefresh) {
-                        this.startBackgroundAddressRefresh(entity);
-                        entity.requestContext.performedBackgroundAddressRefresh = true;
+                    if(!entity.requestContext.isOpenConnectionRequest) { // We will not do address refresh if this is open connection req from CosmosAsyncContainer#openConnectionsAndInitCaches
```

Comment:
Is the goal is to have connections open to real replica's why to exclude?

Reply:
This after the response comes back , we dont want to start address refresh for 410/1002 as this is expected